src/porepy/numerics/ad/equation_system.py

```diff
@@ -423,14 +427,6 @@ def create_variables(
             if var.name == name and var.domain in grids:
                 raise KeyError(f"Variable {name} already defined on {var.domain}.")
 
-        def initialize_data_with_key(data: dict, key: str) -> None:
```

Comment:
Why this change?

Reply:
It was me who originally made this inner function. I made it at the first stage of implementing boundary grids. At that moment, I thought that it would be useful to initialize BG data arrays with the variable names: `data[key][name] = {}`. Now we use separate keys for boundary data, not the variable names, e.g. "bc_data_darcy_flux". Thus, fields with variable names on boundary grids are never used. I didn't see a reason to keep the unified initialization in a separate inner function anymore, so I got rid of it and came back to how it was implemented before me. Do you prefer having the inner function?

Reply:
I think it may make sense to use the (primary or secondary) variable names, i.e, "pressure" and "darcy_flux". Do you agree? In that case, I think keeping the inner function makes sense.

Reply:
This happens inside `EquationSystem.create_variable`, so the field for pressure is created when we're passing name="pressure". "darcy_flux" won't come inside, since we're not creating a primary variable with this name. Or how do you see this?

Reply:
Good point. I still think that we may want to use the same names on the boundary, though.